### PR TITLE
Support for selectors/identifier

### PR DESCRIPTION
**Pull request by marnilSPL**
The idea of this PR is to support the scenario where there are multiple sender instances and the consumer needs to respond to a specific sender.

As a sender you would set the message property ActiveMQBundle.JMS_IDENTIFIER_PROPERTY to some identifier. The consumer would implement receive(Object message, String identifier) and can later send a message using that identifier which the original sender would consume through a selector.

### CODE REVIEW

File: src/main/java/com/kjetland/dropwizard/activemq/ActiveMQReceiver.java [hunk 1]
```diff
@@ -2,5 +2,5 @@
 
 public interface ActiveMQReceiver<T> {
 
-    public void receive(T message);
+    void receive(T message, String identifier);
```
Comment (iulvalSPL) [inline src/main/java/com/kjetland/dropwizard/activemq/ActiveMQReceiver.java]: If you take into account my comments in the service-utils PR, the identifier parameter would have to be replaced by a map of all the message properties. Apparently there is a `message.getPropertyNames()` method available. 😉
Maybe we  could also have a parameter object as the argument: message + properties.

File: src/main/java/com/kjetland/dropwizard/activemq/ActiveMQBundle.java [hunk 2]
```diff
@@ -167,6 +175,12 @@ public ActiveMQSender createSender(String destination, boolean persistent, Optio
     // This must be used during run-phase
     public <T> ActiveMQReceiverHandler<T> registerReceiver(String destination, ActiveMQReceiver<T> receiver, Class<? extends T> clazz,
                                                            ActiveMQBaseExceptionHandler exceptionHandler) {
+        return registerReceiver(destination, receiver, clazz, exceptionHandler, null);
+    }
+
+    // This must be used during run-phase
+    public <T> ActiveMQReceiverHandler<T> registerReceiver(String destination, ActiveMQReceiver<T> receiver, Class<? extends T> clazz,
```
Comment (iulvalSPL) [inline src/main/java/com/kjetland/dropwizard/activemq/ActiveMQBundle.java:181]: we should add an integration test to check the selectors support (smth very similar to what we have in `ActiveMQReceiverHandlerRedeliveryTest`)

**Review by iulvalSPL — CHANGES_REQUESTED** [review]
The selectors support is really useful 💎  but we need some tests for it.
In the service-utils PR I had an optional comment about making the solution more generic. If you don't agree/have time for it, this PR is good to go (as soon as we have tests )

File: src/main/java/com/kjetland/dropwizard/activemq/ActiveMQReceiver.java (hunk 1 above)
Comment (marnilSPL) [reply src/main/java/com/kjetland/dropwizard/activemq/ActiveMQReceiver.java]: I very much agree, at first it wasn't obvious where to "hide" what functionality but I think your suggestion is better. I will move some things around :)

File: src/test/java/com/kjetland/dropwizard/activemq/ActiveMQReceiverHandlerTest.java [hunk 3]
```diff
@@ -78,42 +80,46 @@ public void setUpMocks(List<String> messages) throws Exception {
...
             throw new RuntimeException(THROW_EXCEPTION_IN_CONSUMER);
         }
 
-        if ( THROW_EXCEPTION_IN_CONSUMER_CLOSED.equals(m)) {
+        if (THROW_EXCEPTION_IN_CONSUMER_CLOSED.equals(m)) {
             throw new javax.jms.IllegalStateException("The Consumer is closed");
         }
 
-
+        Vector<String> possibleProperties = new Vector();
```
Comment (iulvalSPL) [inline src/test/java/com/kjetland/dropwizard/activemq/ActiveMQReceiverHandlerTest.java:116]: Good old Vector 😆  it's been a while since I've see it used

**Review by iulvalSPL — APPROVED** [review]
👍

**Pull request merged**